ci/docker/i686-unknown-linux-gnu/Dockerfile

```diff
@@ -1,9 +1,2 @@
-FROM alexcrichton/rust-slave-dist:2015-10-20b
-USER root
-
-WORKDIR /
-RUN curl https://www.cpan.org/src/5.0/perl-5.28.0.tar.gz | tar xzf -
-WORKDIR /perl-5.28.0
-RUN ./configure.gnu
-RUN make -j$(nproc)
-RUN make install
+FROM rust-i686-unknown-linux-gnu
+RUN ln /rustroot/bin/gcc /rustroot/bin/cc
```

Comment:
Is there a nicer, perhaps package-managery way we can have this work?

Reply:
There is no package manager involved here.
Setting environment variable `CC=/rustroot/bin/gcc` should work.

Reply:
No, setting `CC` env doesn't seem to work. Maybe `cc` crate doesn't respect
`CC` env?

Reply:
```console
$ printenv CC CXX
clang
clang++
```

CC is set to clang and we still had the build error.

Reply:
Ah, the target is explicitly configured: https://github.com/rust-lang/rustup.rs/blob/66c4d10871edc307865db031483f2bd799abbba3/ci/run.sh#L9
So it'd be `CC_x86_64-unknown-linux-gnu` for x84_64.

I don't know exact error but if it was the linker then `RUSTFLAGS="-C linker=gcc"` would help but this is getting a bit complex.

Reply:
Is `CC_x86_64-unknown-linux-gnu` standard or it's just used by cc crate?

Reply:
The error message is
```rust
     Running `rustc --crate-name semver_parser /src/target/cargo-home/registry/src/github.com-1ecc6299db9ec823/semver-parser-0.7.0/src/lib.rs --color always --crate-type lib --emit=dep-info,link -C opt-level=3 -C codegen-units=1 -C metadata=c7af5d73bcae4fa9 -C extra-filename=-c7af5d73bcae4fa9 --out-dir /src/target/release/deps -L dependency=/src/target/release/deps --cap-lints allow`
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)
```

Reply:
> Is CC_x86_64-unknown-linux-gnu standard or it's just used by cc crate?

As far as I can tell it's used by `cc-rs` and Rust's bootstrap.

> The error message is

That's the linker error so `RUSTFLAGS="-C linker=gcc"` would work.

Reply:
I still think about it: Why does rustc's docker have no `cc` hard link?
It is common that `cc` hard link to `gcc` on Linux distro.